### DISCUSSION TITLE
What was the primary motivation for creating alpine?

### DISCUSSION THREAD

**Discussion by csenio** [alpinejs/alpine#729] [4. General]
Loving the framework so far, I'm curious what the initial motivation/use case for alpine was. Is it meant for smaller projects like blogs or is it meant as a tool used together with a large backend framework?

**Comment by ryangjchandler** [comment]
@jescowuester Alpine was inspired by Vue primarily, hence why the syntax is so familiar. In terms of usage, it was originally designed for cases where Vue was too heavy, for example, simple modals, dropdowns, tabs, that sort of thing. 

You can use it anywhere, but it does pair really nicely with a server-side framework that has a solid template engine, because that template engine can handle a lot of the componentisation. There are some more front-end based frameworks and template engines that can handle this too, for example, PostHTML.

It's essentially a swiss-army knife for the most common things that you need to do on the front-end of your site or application. Two-way data binding, use `x-model`. Need to do some transitions with display changes, use `x-show` / `x-if` and `x-transition`. The list goes on.
